benchmarks/Microsoft.AspNetCore.Razor.Performance/SyntaxTreeGenerationBenckmark.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using BenchmarkDotNet.Attributes;
+using Microsoft.AspNetCore.Mvc.Razor.Extensions;
+using Microsoft.AspNetCore.Razor.Language;
+
+namespace Microsoft.AspNetCore.Razor.Performance
+{
+    public class SyntaxTreeGenerationBenchmark
+    {
+        public SyntaxTreeGenerationBenchmark()
+        {
+            var current = new DirectoryInfo(AppContext.BaseDirectory);
+            while (current != null && !File.Exists(Path.Combine(current.FullName, "MSN.cshtml")))
+            {
+                current = current.Parent;
+            }
+
+            var root = current;
+            var fileSystem = RazorProjectFileSystem.Create(root.FullName);
+
+            ProjectEngine = RazorProjectEngine.Create(RazorConfiguration.Default, fileSystem, b => RazorExtensions.Register(b)); ;
+
+            MSN = fileSystem.GetItem(Path.Combine(root.FullName, "MSN.cshtml"));
+
+            var directiveFeature = ProjectEngine.EngineFeatures.OfType<IRazorDirectiveFeature>().FirstOrDefault();
+            Directives = directiveFeature?.Directives.ToArray() ?? Array.Empty<DirectiveDescriptor>();
+        }
+
+        public RazorProjectEngine ProjectEngine { get; }
+
+        public RazorProjectItem MSN { get; }
+
+        public DirectiveDescriptor[] Directives { get; }
+
+        [Benchmark(Description = "Razor Design Time Syntax Tree Generation of MSN.com")]
+        public void SyntaxTreeGeneration_DesignTime_LargeStaticFile()
+        {
+            var sourceDocument = RazorSourceDocument.ReadFrom(MSN);
```

Comment:
Should probably lift the reading and options into the preparation phase.

Reply:
Spoke offline. This is what we do for our other benchmarks. So, didn't want to change it here.